### PR TITLE
Add AddToFrameworkScheme function for the test framework

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds a function that allows user to register their custom resources with the test framework's dynamic client and use the dynamic client directly with their custom resources.

Part of issue #387.

### CODE REVIEW

**Comment by AlexNPavel** [comment]
I just removed all the functions that are now unnecessary due to the dynamic client.

**Comment by hasbro17** [comment]
LGTM
defer to @fanminshi

File: test/test-framework/memcached_test.go [hunk 1]
```diff
@@ -15,11 +15,17 @@
 package e2e
 
 import (
+	goctx "context"
+	"fmt"
 	"testing"
 	"time"
 
+	operator "github.com/operator-framework/operator-sdk/test/test-framework/pkg/apis/cache/v1alpha1"
+
```
Comment (fanminshi) [inline test/test-framework/memcached_test.go]: group all `"github.com/operator-framework/operator-sdk/*` together. So no new line here.

File: test/test-framework/memcached_test.go [hunk 2]
```diff
@@ -15,11 +15,17 @@
...
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
 	"github.com/operator-framework/operator-sdk/pkg/util/e2eutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (fanminshi) [inline test/test-framework/memcached_test.go]: new line between line 26 and 27

**Comment by fanminshi** [comment]
lgtm after nits

**Pull request merged**